docs/quickstart.qbk

```diff
@@ -0,0 +1,282 @@
+
+[simplest_hello_world_2_getting_started]
+
+You can also use the `hpx::start` and `hpx::stop` for a non-blocking
+alternative, or use `hpx::resume` and `hpx::suspend` if you need to combine
+__hpx__ with other runtimes.
+
+See __writing_applications__ for more details on how to initialize and run the
+__hpx__ runtime. ]
+
+[heading Writing task-based applications]
+
+So far we haven't done anything that can't be done using the C++ standard
+library. In this section we will give a short overview of what you can do with
+__hpx__ on a single node. The essence is to avoid global synchronization and
+break up your application into small, composable tasks whose dependencies
+control the flow of your application. Remember, however, that __hpx__ allows you
+to write distributed applications in similarly to how you would write
+applications for a single node (see the __intro__).
+
+If you are already familiar with `async` and `future`s from the C++ standard
+library, the same functionality is available in __hpx__.
+
+The following terminology is essential when talking about task-based C++
+programs:
+
+* lightweight thread: Essential for good performance with task-based programs.
+  Lightweight refers to smaller stacks and faster context switching compared to
+  OS-threads. Smaller overheads allow the program to be broken up into smaller
+  tasks, which in turns helps the runtime fully utilize all processing units.
+
+* `async`: The most basic way of launching tasks asynchronously. Returns a
+  `future<T>`.
+
+* `future<T>`: Represents a value of type `T` that will be ready in the future.
+  The value can be retrieved with `get` (blocking) and one can check if the
+  value is ready with `is_ready` (non-blocking).
+
+* `shared_future<T>`: Same as `future<T>` but can be copied (similar to
+  `std::unique_ptr` vs `std::shared_ptr`).
+
+* continuation: A function that is to be run after a previous task has run
+  (represented by a future). `then` is a method of `future<T>` that takes a
+  function to run next. Used to build up dataflow DAGs. `shared_future`s help
+  you split up nodes in the DAG and functions like `when_all` help you join
+  nodes in the DAG.
+
+The following example is a collection of the most commonly used functionality in
+__hpx__.
+
+[import ../examples/quickstart/potpourri.cpp]
+[quickstart_overview]
+
+Try copying the contents to your `main.cpp` file and look at the output. It can
+be a good idea to go through the program step by step with a debugger. You can
+also try changing the types or adding new arguments to functions to make sure
+you can get the types to match. The type of the `then` method can be especially
+tricky to get right (the continuation needs to take the future as an argument).
+
+[note __hpx__ programs accept command line arguments. The most important one is
+`--hpx:threads=N` to set the number of OS-threads used by __hpx__. __hpx__ uses
+all available threads by default. Play around with the example above and see
```

Comment:
Should that be `all available cores by default`?

Reply:
Yes, or even all available processing units.